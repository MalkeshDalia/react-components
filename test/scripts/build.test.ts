import { execSync } from 'child_process';
import * as fs from 'fs';
import * as path from 'path';

describe('build', () => {

  it('build script generates every needed file', () => {
    execSync('npm run build', { stdio:[] })

    const libFiles = fs.readdirSync(path.resolve(__dirname, '../../lib'));
    expect(libFiles).toMatchSnapshot()

    libFiles.forEach(_ => {
      const filePath = path.resolve(__dirname, '../../lib', _);
      if (fs.lstatSync(filePath).isDirectory()) {
        expect(fs.readdirSync(filePath)).toMatchSnapshot();
      }
    })
  })

})
