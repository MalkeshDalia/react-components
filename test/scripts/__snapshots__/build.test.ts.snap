// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build build script generates every needed file 1`] = `
Array [
  "AsyncStatusIndicator/AsyncStatusIndicator.d.ts",
  "AsyncStatusIndicator/AsyncStatusIndicator.js",
  "AsyncStatusIndicator/asyncStatusIndicator.css",
  "AsyncStatusIndicator/index.d.ts",
  "AsyncStatusIndicator/index.js",
  "Badge/Badge.d.ts",
  "Badge/Badge.js",
  "Badge/badge.css",
  "Badge/index.d.ts",
  "Badge/index.js",
  "BrowserDetector/BrowserDetector.d.ts",
  "BrowserDetector/BrowserDetector.js",
  "BrowserDetector/examples.css",
  "BrowserDetector/index.d.ts",
  "BrowserDetector/index.js",
  "Button/Button.d.ts",
  "Button/Button.js",
  "Button/StatefulButton.d.ts",
  "Button/StatefulButton.js",
  "Button/button.css",
  "Button/index.d.ts",
  "Button/index.js",
  "ConfirmationInput/ConfirmationInput.d.ts",
  "ConfirmationInput/ConfirmationInput.js",
  "ConfirmationInput/confirmationInput.css",
  "ConfirmationInput/index.d.ts",
  "ConfirmationInput/index.js",
  "DateField/DateField.d.ts",
  "DateField/DateField.js",
  "DateField/dateField.css",
  "DateField/index.d.ts",
  "DateField/index.js",
  "DatePicker/DatePicker.d.ts",
  "DatePicker/DatePicker.js",
  "DatePicker/datePicker.css",
  "DatePicker/index.d.ts",
  "DatePicker/index.js",
  "Divider/Divider.d.ts",
  "Divider/Divider.js",
  "Divider/divider.css",
  "Divider/index.d.ts",
  "Divider/index.js",
  "Dropdown/Dropdown.d.ts",
  "Dropdown/Dropdown.js",
  "Dropdown/dropdown.css",
  "Dropdown/examples.css",
  "Dropdown/index.d.ts",
  "Dropdown/index.js",
  "DropdownMenu/DropdownMenu.d.ts",
  "DropdownMenu/DropdownMenu.js",
  "DropdownMenu/Menu.d.ts",
  "DropdownMenu/Menu.js",
  "DropdownMenu/dropdownMenu.css",
  "DropdownMenu/examples.css",
  "DropdownMenu/index.d.ts",
  "DropdownMenu/index.js",
  "DropdownMenu/menu.css",
  "FocusableView/FocusableView.d.ts",
  "FocusableView/FocusableView.js",
  "FocusableView/index.d.ts",
  "FocusableView/index.js",
  "FormattedText/FormattedText.d.ts",
  "FormattedText/FormattedText.js",
  "FormattedText/index.d.ts",
  "FormattedText/index.js",
  "Icon/Icon.d.ts",
  "Icon/Icon.js",
  "Icon/index.d.ts",
  "Icon/index.js",
  "Image/Image.d.ts",
  "Image/Image.js",
  "Image/getUrl.d.ts",
  "Image/getUrl.js",
  "Image/index.d.ts",
  "Image/index.js",
  "LoadingSpinner/LoadingSpinner.d.ts",
  "LoadingSpinner/LoadingSpinner.js",
  "LoadingSpinner/index.d.ts",
  "LoadingSpinner/index.js",
  "LoadingSpinner/loadingSpinner.css",
  "Meter/Meter.d.ts",
  "Meter/Meter.js",
  "Meter/examples.css",
  "Meter/index.d.ts",
  "Meter/index.js",
  "Meter/meter.css",
  "MobileDetector/MobileDetector.d.ts",
  "MobileDetector/MobileDetector.js",
  "MobileDetector/index.d.ts",
  "MobileDetector/index.js",
  "Modal/BackgroundDimmer.d.ts",
  "Modal/BackgroundDimmer.js",
  "Modal/Modal.d.ts",
  "Modal/Modal.js",
  "Modal/ModalPortal.d.ts",
  "Modal/ModalPortal.js",
  "Modal/examples.css",
  "Modal/index.d.ts",
  "Modal/index.js",
  "Modal/modal.css",
  "Modal/modalWithContext.d.ts",
  "Modal/modalWithContext.js",
  "MoreOrLess/MoreOrLess.d.ts",
  "MoreOrLess/MoreOrLess.js",
  "MoreOrLess/examples.css",
  "MoreOrLess/index.d.ts",
  "MoreOrLess/index.js",
  "MoreOrLess/moreOrLess.css",
  "NavBar/NavBar.d.ts",
  "NavBar/NavBar.js",
  "NavBar/examples.css",
  "NavBar/index.d.ts",
  "NavBar/index.js",
  "Overflow/Overflow.d.ts",
  "Overflow/Overflow.js",
  "Overflow/index.d.ts",
  "Overflow/index.js",
  "Panel/Panel.d.ts",
  "Panel/Panel.js",
  "Panel/PanelHeader.d.ts",
  "Panel/PanelHeader.js",
  "Panel/TabbedPanel.d.ts",
  "Panel/TabbedPanel.js",
  "Panel/examples.css",
  "Panel/index.d.ts",
  "Panel/index.js",
  "Panel/panel.css",
  "Panel/tabbedPanel.css",
  "Popover/Popover.d.ts",
  "Popover/Popover.js",
  "Popover/examples.css",
  "Popover/index.d.ts",
  "Popover/index.js",
  "ResizeSensor/ResizeSensor.d.ts",
  "ResizeSensor/ResizeSensor.js",
  "ResizeSensor/index.d.ts",
  "ResizeSensor/index.js",
  "Scroll/ScrollView.d.ts",
  "Scroll/ScrollView.js",
  "Scroll/easingFunctions.d.ts",
  "Scroll/easingFunctions.js",
  "Scroll/index.d.ts",
  "Scroll/index.js",
  "ScrollView/ScrollView.d.ts",
  "ScrollView/ScrollView.js",
  "ScrollView/index.d.ts",
  "ScrollView/index.js",
  "ScrollView/scrollView.css",
  "Tablo/Cell",
  "Tablo/Column",
  "Tablo/ColumnGroup",
  "Tablo/Footer",
  "Tablo/Header",
  "Tablo/Tablo.d.ts",
  "Tablo/Tablo.js",
  "Tablo/index.d.ts",
  "Tablo/index.js",
  "Tablo/patch-fixed-data-table-2.d.ts",
  "Tablo/patch-fixed-data-table-2.js",
  "Tablo/plugins",
  "Tablo/tablo.css",
  "Tablo/tabloVariables.css",
  "Tablo/utils.d.ts",
  "Tablo/utils.js",
  "TextOverflow/TextOverflow.d.ts",
  "TextOverflow/TextOverflow.js",
  "TextOverflow/TextOverflowTouch.d.ts",
  "TextOverflow/TextOverflowTouch.js",
  "TextOverflow/index.d.ts",
  "TextOverflow/index.js",
  "TextOverflow/textOverflowTouch.css",
  "TimePicker/TimePicker.d.ts",
  "TimePicker/TimePicker.js",
  "TimePicker/index.d.ts",
  "TimePicker/index.js",
  "Toaster/TimerToast.d.ts",
  "Toaster/TimerToast.js",
  "Toaster/Toaster.d.ts",
  "Toaster/Toaster.js",
  "Toaster/index.d.ts",
  "Toaster/index.js",
  "Toggle/Toggle.d.ts",
  "Toggle/Toggle.js",
  "Toggle/index.d.ts",
  "Toggle/index.js",
  "Toggle/toggle.css",
  "Tooltip/Tooltip.d.ts",
  "Tooltip/Tooltip.js",
  "Tooltip/TooltipTouch.d.ts",
  "Tooltip/TooltipTouch.js",
  "Tooltip/index.d.ts",
  "Tooltip/index.js",
  "Tooltip/tooltip.css",
  "Tooltip/tooltipTouch.css",
  "TransitionWrapper/TransitionWrapper.d.ts",
  "TransitionWrapper/TransitionWrapper.js",
  "TransitionWrapper/index.d.ts",
  "TransitionWrapper/index.js",
  "index.d.ts",
  "index.js",
  "utils/index.d.ts",
  "utils/index.js",
  "utils/log.d.ts",
  "utils/log.js",
  "utils/theme",
]
`;
